daemon/swap_daemon_erc20_test.go

```diff
@@ -21,26 +24,52 @@ import (
 // Tests the scenario where Bob has XMR and enough ETH to pay gas fees for the token claim. He
 // exchanges 2 XMR for 3 of Alice's ERC20 tokens.
 func TestRunSwapDaemon_ExchangesXMRForERC20Tokens(t *testing.T) {
-	minXMR := coins.StrToDecimal("1")
-	maxXMR := coins.StrToDecimal("2")
-	exRate := coins.StrToExchangeRate("1.5")
+	fundingEC := extethclient.CreateTestClient(t, tests.GetTakerTestKey(t))
```

Comment:
I refactored this test to make sure that Alice uses a completely new ethereum key. At the time, I was trying to figure out if the failures were related to `approve`. I think the test is better this way, so I didn't revert the changes.